tests/business/notebookrunner_test.py

```diff
@@ -59,7 +59,7 @@ async def test_run(
                     "options": {
                         "spawn_settle_time": 0,
                         "execution_idle_time": 0,
-                        "max_executions": 1,
+                        "max_executions": 10,
```

Comment:
I think this would be a stronger test if `max_executions` exactly matched the number of notebooks that should be run, since then you can also test for off-by-one errors (make sure the first notebook isn't run twice, for instance).

Reply:
Yeah, my change makes that test weaker, but I also wanted to ensure that no notebooks from excluded dirs would be run.  I changed `max_execution` back to the exact number of notebooks, but I also moved the "cycle done" message from the beginning of the _next_ cycle, to the end of `execute_code`, and I'm asserting that's logged.

That gets us the best of both worlds, right?